### PR TITLE
Token rollover

### PR DESCRIPTION
**Pull request by cornelinux**
This is an alternative implementation of the token rollover.
It uses the normal insite token enrollment and also
allows to change data (key length, otp length) if the administrator
allows so.

Still missing the webui policy.

Working on #2613

### CODE REVIEW

**Comment by cornelinux** [comment]
@laclaro I would like to discuss this.

File: doc/policies/webui.rst [hunk 1]
```diff
@@ -300,6 +300,20 @@ being disabled.
 
 (Since privacyIDEA 3.0)
 
+token_rollover
+~~~~~~~~~~~~~~
+
+type: str
+
+This is a whitespace separated list of tokentypes, for which a rollover button is
+displayed in the token details and in the token list. This button will generate a
```
Comment (laclaro) [inline doc/policies/webui.rst:309]: this PR leaves the token list untouched.

File: privacyidea/lib/policy.py [hunk 2]
```diff
@@ -2381,6 +2382,13 @@ def get_static_policy_definitions(scope=None):
                 'desc': _("The tokenwizard will be displayed in the token "
                           "menu, even if the user already has a token.")
             },
+            ACTION.TOKENROLLOVER: {
+                'type': 'str',
+                'desc': _('This is a whitespace separated list of tokentypes, '
+                          'for which a rollover button is displayed in the token '
+                          'details and in the token list.'),
```
Comment (laclaro) [inline privacyidea/lib/policy.py:2389]: see above. Please remove token list.

**Review by laclaro — CHANGES_REQUESTED** [review]
Please remove the hints to a button in the token list as we do not implement it with this PR.

**Review by laclaro — APPROVED** [review]
I quickly corrected the descriptions.

**Pull request merged**